app/src/androidTest/java/com/github/sdp_begreen/begreen/fragments/SendPostFragmentTest.kt

```diff
@@ -178,10 +182,19 @@ class SendPostFragmentTest {
     }
 
     @Test
-    fun typeInCategoryWorks() {
-        // Check the category input
-        onView(withId(R.id.post_category)).perform(typeText("test"))
-        onView(withId(R.id.post_category)).check(matches(withText("test")))
+    fun changeTrashCategoryWorks() {
+
+        // Pick a trash category at random
+        onView(withId(R.id.post_trash_category_selector)).perform(click())
```

Comment:
I think that choosing the first trash category was enough for a test, but it is true that choosing it randomly improves the "robustness" of the test. As you have complicated your life, wouldn't you think it's better to loop through all elements ? I feel like it would have been easier, and more robust. Let me know what you think about it, but your current implementation is already good.

Reply:
I think chosing a new value once and checking that it is correctly displayed is enough